src/main/java/uk/gov/hmcts/darts/common/repository/EventRepository.java

```diff
@@ -104,4 +106,13 @@ HAVING count(event_id) > 1
         """)
     void updateAllEventIdEventsToNotCurrentWithTheExclusionOfTheCurrentEventPrimaryKey(
         List<Integer> eventIdsPrimaryKeysLst, List<Integer> eventIdLst);
+
+    @EntityGraph(attributePaths = {"id", "timestamp", "eventId", "messageId", "eventText"})
+    @Query("""
+        SELECT ee
+        FROM EventEntity ee
+        WHERE ee.timestamp >= :startDateTime
+        AND ee.timestamp <= :endDateTime
+        """)
+    List<EventEntity> findAllForDuplicateProcessing(OffsetDateTime startDateTime, OffsetDateTime endDateTime);
```

Comment:
Maybe a more generic method name would be better, as stricly speaking the query doesn't care about dupes. E.g. `findAllBetweenDateTimesInclusive()`